controller/share_manager_controller.go

```diff
@@ -1050,6 +1050,50 @@ func (c *ShareManagerController) getShareManagerTolerationsFromStorageClass(sc *
 	return tolerations
 }
 
+func (c *ShareManagerController) cleanupServiceAndEndpoint(shareManager *longhorn.ShareManager) error {
+	service, err := c.ds.GetService(c.namespace, shareManager.Name)
+	if err != nil {
+		// if NotFound, means the service/endpoint is already cleaned up
```

Comment:
I'm thinking of removing the cleanup from the setting controller so it can be handled just here instead. What do you think?

Reply:
After discussing this offline with @c3y1huang, we will add the status check here and remove the cleanup from the setting controller.

I will mark it ready once I complete the above changes. Thanks!

Reply:
Updated! Please help to review. Thanks!